### PR TITLE
Add family share

### PR DESCRIPTION
**Pull request by mlajx**
First of all, sorry for my english, it is not my native language.

I'm creating a project on my own and ended up needing authorization and removal of devices and borrowers.

So I ended up studying your code and implemented those functions in it.

### CODE REVIEW

**Comment by Revadike** [comment]
Resolves https://github.com/DoctorMcKay/node-steam-user/issues/267 🎉

File: components/device.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
+const StdLib = require('@doctormckay/stdlib');
+const Helpers = require('./helpers.js');
+const SteamUser = require('../index.js');
+
+/**
+ * Get authorized devices.
+ * @param {boolean} [inclTokens=false] - If true, include canceled devices
```
Comment (Revadike) [inline components/device.js:7]: `inclTokens`? Also, while `canceled` isn't technically incorrect, usually `cancelled` is before.

File: components/device.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+/**
+ * Authorize local device.
+ * @param {string} deviceName
+ */
+SteamUser.prototype.authorizeLocalDevice = function(deviceName){
+	if(!deviceName){
+		throw new Error('The \'deviceName\' argument is required.');
+	}
+
+	this._send(SteamUser.EMsg.ClientAuthorizeLocalDeviceRequest, {
```
Comment (Revadike) [inline components/device.js:39]: No promise/callback here?

**Comment by Revadike** [comment]
Can you also authorize a remote machineid?

File: components/device.js (hunk 1 above)
Comment (mlajx) [reply components/device.js]: oh ... sorry hehe. Anyway, I fixed the "inclTokens" and changed the "canceled".

File: components/device.js (hunk 2 above)
Comment (mlajx) [reply components/device.js:39]: When "ClientAuthorizeLocalDeviceRequest" is completed, an "authorizedLocalDevice" event is triggered

the same goes for "ClientDeauthorizeDeviceRequest"

both returns the device.

then you can use

```js
client.on('authorizedLocalDevice', function (device) {
  // console.log(device);
}
```

**Comment by mlajx** [comment]
> Can you also authorize a remote machineid?

I haven't found anything that I can remotely authorize a machine on, just locally.

About the documentation ...
I am literally terrible in English and will probably have a lot of wrong writing.
So, I believe that someone who has English as a native language, would be able to write much better.

**Comment by Revadike** [comment]
There seems to be methods for remote devices as well: https://github.com/dootdoot/openstreamingclient2/blob/master/Raw%20protobufs/steammessages_deviceauth.steamclient.proto#L101

Could you try to implement them as well?

**Pull request merged**